packages/x-scheduler/src/material/event-calendar/EventCalendar.test.tsx

```diff
@@ -96,97 +98,186 @@ describe('EventCalendar', () => {
+      expect(screen.queryByRole('columnheader', { name: /Saturday/i })).to.equal(null);
+
+      // Show the weekends again
+      await openPreferencesMenu(user);
+      await toggleShowWeekends(user);
+      await user.click(document.body);
+
+      expect(screen.getByRole('columnheader', { name: /Sunday/i })).not.to.equal(null);
+      expect(screen.getByRole('columnheader', { name: /Saturday/i })).not.to.equal(null);
+    });
 
-    expect(await findWeekHeaders()).to.have.lengthOf(0);
+    it('should allow to show / hide the weekends using the UI in the agenda view', async () => {
+      const { user } = render(<EventCalendar events={[]} defaultView="agenda" />);
+
+      // Weekends should be visible by default
+      expect(screen.getByLabelText(/Saturday 31/i)).not.to.equal(null);
+      expect(screen.getByLabelText(/Sunday 1/i)).not.to.equal(null);
+
+      // Hide the weekends
+      await openPreferencesMenu(user);
+      await toggleShowWeekends(user);
+      await user.click(document.body);
+
+      expect(screen.queryByLabelText(/Saturday 31/i)).to.equal(null);
+      expect(screen.queryByLabelText(/Sunday 1/i)).to.equal(null);
+
+      // Show the weekends again
+      await openPreferencesMenu(user);
+      await toggleShowWeekends(user);
+      await user.click(document.body);
+
+      expect(screen.getByLabelText(/Saturday 31/i)).not.to.equal(null);
+      expect(screen.getByLabelText(/Sunday 1/i)).not.to.equal(null);
+    });
+
+    it('should allow to show / hide the week number using the UI in the month view', async () => {
+      const { user } = render(<EventCalendar events={[]} defaultView="month" />);
+
+      // Week number should not be visible by default
+      const findWeekHeaders = () => screen.queryAllByRole('rowheader', { name: /week/i });
+      expect(await findWeekHeaders()).to.have.lengthOf(0);
+
+      // Show the week number
+      await openPreferencesMenu(user);
+      await toggleShowWeekNumber(user);
+      await user.click(document.body);
+
+      expect(await findWeekHeaders()).to.have.lengthOf.above(0);
+
+      // Hide the week number again
+      await openPreferencesMenu(user);
+      await toggleShowWeekNumber(user);
+      await user.click(document.body);
+
+      expect(await findWeekHeaders()).to.have.lengthOf(0);
+    });
+
+    it('should allow to change the time format using the UI in the week view', async () => {
+      const { user } = render(<EventCalendar events={[]} />);
+
+      // AM/PM format should be visible by default
+      expect(screen.queryAllByText(/AM|PM/).length).to.be.above(0);
+
+      // Disable AM/PM format
+      await openPreferencesMenu(user);
+      await changeTo24HoursFormat(user);
+      await user.click(document.body);
+
+      expect(screen.queryAllByText(/AM|PM/).length).to.equal(0);
+
+      // Show AM/PM format again
+      await openPreferencesMenu(user);
+      await changeToAmpmFormat(user);
```

Comment:
I would prefer to say "12h / 24" everywhere instead of "AM/PM / 24h".
Seems more consistent to me, wdyt? (same for the label)

```suggestion
      await changeTo12HoursFormat(user);
```